homework-06/main.js

```diff
@@ -0,0 +1,115 @@
+// 1
+function compareNumbers(num1, num2, num3, num4) {
+    if ((Number.isNaN(num1)) || (Number.isNaN(num2)) || (Number.isNaN(num3)) || (Number.isNaN(num4))) {
+        throw 'Parameter is not a number!';
+    }
+
+    let result;
+    if (num1 < num2) {
+        result = num1;
+    } else {
+        result = num2;
+    }
+
+    let result1;
+    if (num3 < num4) {
+        result1 = num3;
+    } else {
+        result1 = num4;
+    }
+
+    if (result < result1) {
+        return result;
+    } else {
+        return result1;
+    }
+}
+console.log(compareNumbers(12, 2, 4, -5));
+
+const lesserNumber = function (num1, num2, num3, num4) {
+    if ((Number.isNaN(num1)) || (Number.isNaN(num2)) || (Number.isNaN(num3)) || (Number.isNaN(num4))) {
+        throw 'Parameter is not a number!';
+    }
+    return Math.min(num1, num2, num3, num4);
+}
+console.log(lesserNumber(2, 3, -7, 8));
+
+// 2
+function greatestCommonDivisor(a, b) {
+    if ((Number.isNaN(a)) || (Number.isNaN(b))) {
+        throw 'Parameter is not a number!';
+    }
+    if (!Number.isInteger(a) || (!Number.isInteger(b))) {
+        throw 'Parameter is not an integer number!';
+    }
+    if ((a < 0) || (b < 0)) {
+        throw 'Parameter is not a positive number!'
+    }
+
+    let n = Math.min(a, b);
+    for (let i = n - 1; i > 1; i--) {
+        if ((a % i == 0) && (b % i == 0)) {
+            return i;
+        }
+    }
+    return 'This numbers don\'t have common divisor';
+}
+console.log(greatestCommonDivisor(45, 15))
+
+// 3
+const isPerfectNumber = function (n) {
+    if (Number.isNaN(n)) {

```

Comment:
And here
<img width="566" alt="image" src="https://user-images.githubusercontent.com/84010501/191210447-09a64dc4-57c7-4c9c-a8e6-6e5d3367aed0.png">